src/components/Dashboard/ChangingPlans.component.js

```diff
@@ -51,20 +52,7 @@ export function ChangingPlansComponent() {
 
 	useEffect(() => {
 		getPlans().then(plans => {
-			plans.sort((a, b) => {
-				// Extract the numbers from the plan names
-				const numA = parseInt(a.name.match(/\d+/)?.[0] || '0')
-				const numB = parseInt(b.name.match(/\d+/)?.[0] || '0')
-
-				// Compare the numbers
-				if (numA !== numB) {
-					return numA - numB
-				}
-
-				// If the numbers are the same, compare the names
-				return b.name.localeCompare(a.name)
-			})
-			setPlans(plans)
+			sortAndSetPlans(plans, setPlans)
```

Comment:
en rentrant le setPlan dans la fonction de tris, tu melange les responsabilité de la fonctions, et tu cache le mechanisme de changement d'etat :thinking:  c'est pas une bonne pratique je pense, faudrait faire un 
```suggestion
			setPlans(plans.map(sortAndSetPlans))
```